qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -216,6 +230,19 @@ def get_solver(  # type: ignore[override]
         ansatz.num_spin_orbitals = num_spin_orbitals
         ansatz.initial_state = initial_state
 
+        initial_point = self.initial_point
+        if initial_point is not None and not isinstance(initial_point, np.ndarray):
+            # UCC ansatz must be built early to compute excitation list.
+            ansatz._build()
```

Comment:
Currently, one needs to call `_build()` as the existing UCC code only computes its excitation list when it builds the excitation operators. Should UCC be refactorized to try to separate the excitation_list computation and the rest of the build?